andela/Ah-backend-xmen
# #162725260 Users should be able to rate an article

#### What does this PR do?
This PR adds functionality for articles to be rated by readers. And In case one fetches an article or all articles, he/she should be able to see the average ratings of the article.

#### Description of Task to be completed?
##### Modify fetch articles endpoint to have 
```
{
      "average_rating": <average of the ratings scored>
}
```
##### Implement an endpoint
- Rate an article
`POST/api/articles/<slug>/rate`

#### How should this be manually tested?
- First things first, fetch this branch by running the command `git fetch origin ft-rate-article-162725260` in the terminal.
- Check into this branch by running `git checkout ft-rate-article-162725260`
- Activate your virtual environment by running `source venv/bin/activate`
- Drop your current database and create a new one by running `create database <database name>;` and `drop database <database name>;` in the psql shell
- Apply the new changes to the models by running `python manage.py migrate`
- Start the application by running `python manage.py runserver`
- Click the `Run in postman` button to test the endpoint as shown in the screenshots below

#### What are the relevant pivotal tracker stories?
[#162725260](https://www.pivotaltracker.com/story/show/162725260)

#### Screenshots

#### Rate an Article
![image](https://user-images.githubusercontent.com/36506567/51764019-0c736700-20e5-11e9-85cb-36865f323bde.png)

#### Already Rated
![image](https://user-images.githubusercontent.com/36506567/51763807-793a3180-20e4-11e9-893a-174e67562e9d.png)

#### Average Rating
![image](https://user-images.githubusercontent.com/36506567/51764952-886eae80-20e7-11e9-8753-698164887b50.png)

#### POSTMAN
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/b7f9d4812b20fd57fe74)
